mlflow/tracking/fluent.py

```diff
@@ -69,9 +70,29 @@
     import numpy
     import PIL
 
-_active_run_stack = []
-_active_experiment_id = None
-_last_active_run_id = None
+
+class DefaultSetContextVar:
+    def __init__(self, contextvar, default: Callable):
+        self.contextvar = contextvar
+        self.default = default
+
+    def get(self):
+        try:
+            return self.contextvar.get()
+        except LookupError:
+            if self.default is not None:
+                self.set(self.default())
```

Comment:
If no default value we should raise error here?

Reply:
If there's no default value you'll get a `LookupError` as per normal with contextvars when the next line tries `self.get()`. Could also just rethrow the original LookupError here to avoid wasting time calling `.get()`, maybe that's better